utils/economic-checks/src/targets.rs

```diff
@@ -0,0 +1,703 @@
+            rng.fill_bytes(&mut bytes);
+
+            let u = Unstructured::new(&bytes[..]);
+
+            // Generate messages queue
+            let queue_len = params.queue_len as usize;
+            let mut queue = BTreeMap::<u16, Vec<(u16, Seed)>>::new();
+
+            u.arbitrary_take_rest_iter::<(u16, u16, [u8; 32])>()
+                .map_err(|_| {
+                    DispatchError::from("Failed to draw random tuple from unstructured source")
+                })?
+                .take(queue_len)
+                .for_each(|v| {
+                    let (block_num, contract_num, seed) = v.unwrap(); // Guaranteed to have value
+                    queue
+                        .entry(block_num % (MAX_BLOCK - 2) + MIN_BLOCK) // [MIN_BLOCK..MAX_BLOCK]
+                        .or_insert(Vec::new())
+                        .push((contract_num % num_hashes, Seed(seed)));
+                });
+
+            log::debug!(
+                "Queue: num entries = {}, num blocks = {}, queue: {:?}",
+                queue_len,
+                queue.len(),
+                queue,
+            );
+
+            let mut current_block = 2_u16;
+
+            // check the first block to send a message in is greater than current block
+            let mut blocks = queue.keys().cloned();
+            if let Some(first_block) = blocks.next() {
+                assert!(first_block > current_block);
+            }
+
+            // for a block number in [3..N] send out messages and run queue processing
+            for (blk, entries) in queue {
+                run_to_block_with_ocw(blk as u32, pool.clone(), None);
+
+                for (hash_id, seed) in entries {
+                    let author = &accounts
+                        .choose(&mut rng)
+                        .expect("Accounts vec must not be empty")
+                        .0;
+
+                    // Decide whether the message should contain value
+                    let value = match rng.next_u32() >> 28 {
+                        0_u32 => rng.next_u32() as u128, // 1/16 chances to have value
+                        _ => 0_u128,
+                    };
+                    Gear::send_message(
+                        Origin::signed(author.clone()),
+                        hashes[hash_id as usize],
+                        seed.0.to_vec(),
+                        params.gas_limit,
+                        value,
+                    )
+                    .map_err(|e| e.error)?;
+                }
+                current_block = blk;
+
+                // assert_eq!(total_reserved_balance(), <Runtime as pallet_gear::Config>::GasHandler::total_supply() as u128);
+                // assert_eq!(total_gas_in_wait_list() as u128, total_reserved_balance());
+                // assert_eq!(
+                //     total_gas_in_wait_list(),
+                //     <Runtime as pallet_gear::Config>::GasHandler::total_supply()
+                // );
```

Comment:
dead code

Reply:
not completely dead: the condition needs to be reworked because formulated as is it sometimes triggers false positives. It's rather a TODO